### ISSUE TITLE
Create graphical frontend to testdisk

### ISSUE THREAD

**Issue by shasheene** [rescuezilla/rescuezilla#12]
A [user on the Redo Backup and Recovery forum in 2011](https://sourceforge.net/p/redobackup/discussion/1169664/thread/110ce8c6/) struggled to save recovered files onto their hard drive.

A graphical wizard for testdisk would be generally useful and reduce support overhead.

**Comment by Tarek-Hasan** [comment]
Any update on this work?
Also official Testdisk GUI 'QPhotoRec' doesnot have file formats categorized, like video-mp4, mkv, mpg etc. Its a mess out there! It would be easier for people if file formats are categorized under similar types, like audio, document, video etc.
If you need help with categorizing I am available.
And thank you for this great project.

**Comment by shasheene** [comment]
I agree this is a program that somebody really needs to develop, but I am not intending on working on this issue for at least 12 months. There's a lot of work that does into design, implementing and maintaining such software  that needs to be tested on a large number of hard drives etc.

By the way, as you may know TestDisk and PhotoRec are closely related, but they are very different applications: TestDisk is a versatile file undelete application, and PhotoRec is used for deep-scanning for photographs from corrupted memory cards.

Rescuezilla has the command-line TestDisk (and PhotoRec) applications available from the Start Menu. They're definitely very finicky to use, but TestDisk is _very_ powerful.

**Comment by regs01** [comment]
I may suggest doing TestDisk from sources. This will bring QPhotoRec.

For Jammy it needs next packages on host to compile
```
sudo apt install build-essential libext2fs-dev libewf-dev libncurses5-dev libncursesw5-dev ntfs-3g-dev libjpeg-dev uuid-dev zlib1g-dev dh-autoreconf git
sudo apt install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools qttools5-dev-tools qtdeclarative5-dev pkg-config
```

Compilation. Compiled `testdisk`, `photorec`, `qphotorec`, `fidentify` will  appear in src folder.
```
git clone https://git.cgsecurity.org/testdisk.git
cd testdisk
mkdir config
autoreconf --install -W all -I config
./configure
make
```

To run on Live CD it needs next packages
```
libqt5core5a 
libqt5gui5 
libqt5widgets5 
libewf2
```

**Comment by shasheene** [comment]
I'm not a big fan of QPhotoRec's overly complicated interface. But there is a strong case to be made for including it until I develop a TestDisk GUI.

Also while PhotoRec and TestDisk are similar applications created by the same author, as you may know they each have very different behavior.

Rescuezilla's start-menu entries for PhotoRec as "photograph deep scan" and TestDisk as "file undelete".

**Comment by Tarek-Hasan** [comment]
Will it be possible to include QPhotoRec in Rescuezilla until a better solution is available?

**Comment by ShlomiRex** [comment]
> I may suggest doing TestDisk from sources. This will bring QPhotoRec.
> 
> For Jammy it needs next packages on host to compile
> 
> ```
> sudo apt install build-essential libext2fs-dev libewf-dev libncurses5-dev libncursesw5-dev ntfs-3g-dev libjpeg-dev uuid-dev zlib1g-dev dh-autoreconf git
> sudo apt install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools qttools5-dev-tools qtdeclarative5-dev pkg-config
> ```
> 
> Compilation. Compiled `testdisk`, `photorec`, `qphotorec`, `fidentify` will appear in src folder.
> 
> ```
> git clone https://git.cgsecurity.org/testdisk.git
> cd testdisk
> mkdir config
> autoreconf --install -W all -I config
> ./configure
> make
> ```
> 
> To run on Live CD it needs next packages
> 
> ```
> libqt5core5a 
> libqt5gui5 
> libqt5widgets5 
> libewf2
> ```
> 
> <img alt="VirtualBoxVM_2022-05-20_03-56-15" width="1440" src="https://user-images.githubusercontent.com/10468217/169427787-02df4d4a-c213-4160-9c06-42abd98333dd.png">

What is the desktop environment / distrobution of linux you having? its lit bro 💯

**Comment by shasheene** [comment]
I've played around with QPhotoRec a long time ago, and there's a strong case for including it Rescuezilla as a stop-gap. Particularly since PhotoRec and TestDesk themselves are already included in there terminal user-interface versions.

But I'm not satisfied with QPhotoRec's user-experience / user-interface. I think there can be a much more usable frontend, and I might have more to share in a few weeks.

By the way, as you may know the terminal user-interface for PhotoRec and TestDesk internally operate differently (one photograph deep scan the other is undeleting files).

I'll play around with QPhotoRec again and may include it if I don't have my own TestDisk frontend developed soon.

**Comment by shasheene** [comment]
> What is the desktop environment / distrobution of linux you having? its lit bro :100: 

Rescuezilla's live environment is currently based on Ubuntu, and a Debian version is intended too.

The general look-and-feel of Rescuezilla's desktop environment can be mostly traced back to the LXDE desktop environment. **Specifically the "lxpanel" program**, and to some degree the underlying openbox window manager.

The start menu icon and a few colors of the menu bar were selected to imitate an old version of a different tool named `adeskbar` used by Redo Backup and Recovery 1.0.4 (which was not available when I started developing Rescuezilla, but I wanted to maintain consistency of look-and-feel to that tool).

Rescuezilla doesn't use the entirety of LXDE, just a few pieces.

**Comment by Tarek-Hasan** [comment]
I recently found out, [RecoveRx](https://www.transcend-info.com/support/software/recoverx) by Transcend is just an alternative GUI of TestDisk. [Source code](https://tw.transcend-info.com/Software/1721/) released under MS-PL license.

By simply adding a method to categorizing file formats under similar types, like audio, document, video etc. in QPhotoRec I believe you can improve its usability significantly, similar to RecoveRX.

**Comment by Y-PLONI** [comment]
I’m wondering if there’s any plan to create a user interface (UI) for TestDisk? I have some knowledge of computers, but working with command-line tools feels a bit intimidating for me.

I’m not a programmer, so I’m curious: is creating a UI for TestDisk really that complex?

**Comment by Tarek-Hasan** [comment]
There's a new data recovery tool - [RevLix](https://github.com/cektor/RevLix), which is a GUI for PhotoRec, TestDisk, ddrescue, foremost, and S.M.A.R.T. analysis. Though, it currently doesn't have a robust selection method like RecoveRX, still it could be a great addition to Rescuzilla.
